tests/cpp/include/test_legacy_op.h

```diff
@@ -0,0 +1,551 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file test_op.h
+ * \brief operator unit test utility functions
+ * \author Chris Olivier
+ *
+ * These classes offer a framework for developing, testing and debugging operators
+ * in C++.  They work for both CPU and GPU modes, as well as offer a timing
+ * infrastructure in order to test inidividual operator performance.
+ *
+ * Operator data can be validated against general logic,
+ * stored scalar values (which can be generated by this code from an existing operator via
+ * BasicOperatorData::dumpC(), as well as against each other (ie check that
```

Comment:
Much of this code was moved from test_op.h and then ported to use the new templating scheme.  The general flow remains similar.